lib/requestability_resolver.js

```diff
@@ -2,46 +2,57 @@ const DeliveryLocationsResolver = require('./delivery-locations-resolver')
       .forEach((hit) => {
+        const parentBibHasFindingAid = !!hit._source.supplementaryContent?.find((el) => el.label.toLowerCase() === 'finding aid')
         hit._source.items = hit._source.items.map((item) => {
           if (item.electronicLocator) return item
           let deliveryInfo
           const itemIsInRecap = isInRecap(item)
           let physRequestableCriteria
-          const hasRecapCustomerCode = item.recapCustomerCode && item.recapCustomerCode[0]
+          const hasRecapCustomerCode = item.recapCustomerCode?.[0]
           if (itemIsInRecap) {
             // recap items missing codes should default to true for phys and edd
             // requestable, unless it has a non-requestable holding location
             deliveryInfo = DeliveryLocationsResolver.getRecapDeliveryInfo(item)
             physRequestableCriteria = hasRecapCustomerCode
-              ? `${(deliveryInfo.deliveryLocation &&
-                deliveryInfo.deliveryLocation.length) || 0} delivery locations.`
+              ? `${(deliveryInfo.deliveryLocation?.length) || 0} delivery locations.`
               : 'Missing customer code'
           } else if (!itemIsInRecap) {
             deliveryInfo = DeliveryLocationsResolver.getOnsiteDeliveryInfo(item)
-            physRequestableCriteria = `${(deliveryInfo.deliveryLocation &&
-              deliveryInfo.deliveryLocation.length) || 0} delivery locations.`
+            physRequestableCriteria = `${(deliveryInfo.deliveryLocation?.length) || 0} delivery locations.`
           }
-          item.eddRequestable = !!deliveryInfo.eddRequestable
-          item.physRequestable = !!(deliveryInfo.deliveryLocation &&
-            deliveryInfo.deliveryLocation.length)
-          item.specRequestable = !!item.aeonUrl
+          item.specRequestable = this.buildSpecRequestable(item, parentBibHasFindingAid)
+          item.physRequestable = !!deliveryInfo.deliveryLocation?.length
+          item.eddRequestable = !!deliveryInfo.eddRequestable && !item.specRequestable
           // items without barcodes should not be requestable
           const hasBarcode = (item.identifier || []).some((identifier) => /^(urn|bf):[bB]arcode:\w+/.test(identifier))
           if (isItemNyplOwned(item) && !hasBarcode) {
             physRequestableCriteria = 'NYPL item missing barcode'
             item.physRequestable = false
           }
+          if (item.specRequestable && item.physRequestable) {
+            item.physRequestable = false
+            physRequestableCriteria = 'specRequestable overrides physRequestable location'
+          }
           logger.debug(`item ${item.uri}: `, { physRequestable: item.physRequestable, physRequestableCriteria })
           item.requestable = [item.eddRequestable || item.physRequestable || item.specRequestable]
+          console.log({ specRequestable: item.specRequestable, physRequestable: item.physRequestable, eddRequestable: item.eddRequestable, physRequestableCriteria })
```

Comment:
Whoops - we should remove this upstream given it's fired for every item in the response.

Reply:
oof good catch.

Reply:
the console log is removed in [this pr](https://github.com/NYPL/discovery-api/pull/436)